bazel/foreign_cc/nghttp2.patch

```diff
@@ -15,3 +15,73 @@ index 35c77d1d..47bd63f5 100644
  endif()
  # AC_TYPE_UINT8_T
  # AC_TYPE_UINT16_T
+# https://github.com/nghttp2/nghttp2/pull/1468
+diff --git a/lib/nghttp2_buf.c b/lib/nghttp2_buf.c
+index 2a435bebf..92f97f7f2 100644
+--- a/lib/nghttp2_buf.c
++++ b/lib/nghttp2_buf.c
+@@ -82,8 +82,10 @@ void nghttp2_buf_reset(nghttp2_buf *buf) {
+ }
+ 
+ void nghttp2_buf_wrap_init(nghttp2_buf *buf, uint8_t *begin, size_t len) {
+-  buf->begin = buf->pos = buf->last = buf->mark = begin;
+-  buf->end = begin + len;
++  buf->begin = buf->pos = buf->last = buf->mark = buf->end = begin;
++  if (buf->end != NULL) {
```

Comment:
Can we land this one first? I'm pretty unenthusiastic about these patches, they make upgrades really painful.

Reply:
It is already landed but not in a stable release: https://github.com/nghttp2/nghttp2/pull/1468#issuecomment-639523675, do you want to bump nghttp2 instead of patching?

Reply:
Since this lands right after the 1.41 release and 1.42 start, I think we can just bump.

Reply:
@htuch those commit lands on master breaks Windows, let's merge this with patch?

Reply:
Ack, the saving grace here is that any nghttp2 bump will require us to get rid of most of these. @wrowe is there a plan to fix the nghttp2 Windows support? We don't want to maintain Windows patches indefinitely here.